src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -531,6 +533,39 @@ public async Task RestoreDatabase()
                 if (ResourceNameValidator.IsValidResourceName(databaseName, ServerStore.Configuration.Core.DataDirectory.FullPath, out string errorMessage) == false)
                     throw new BadRequestException(errorMessage);
 
+                var extension = Path.GetExtension(restoreConfigurationJson.LastFileNameToRestore);
+                if (extension == Constants.Documents.PeriodicBackup.SnapshotExtension || extension == Constants.Documents.PeriodicBackup.EncryptedSnapshotExtension)
+                {
+                    long backupSizeInBytes;
+                    try
+                    {
+                        using (var zip = ZipFile.OpenRead(Path.Combine(restoreConfigurationJson.BackupLocation, restoreConfigurationJson.LastFileNameToRestore)))
+                            backupSizeInBytes = zip.Entries.Sum(entry => entry.Length);
+                    }
+                    catch
+                    {
+                        throw new IOException($"Could not open the backup file, provided file : {Path.Combine(restoreConfigurationJson.BackupLocation, restoreConfigurationJson.LastFileNameToRestore)} Please provide a valid backup file.");
+                    }
+
+                    var destinationPath = string.IsNullOrWhiteSpace(restoreConfigurationJson.DataDirectory)
+                        ? Path.Combine(ServerStore.Configuration.Core.DataDirectory.FullPath, restoreConfigurationJson.DatabaseName)
```

Comment:
instead of doing that you should use `ServerStore.Configuration.Core.DataDirectory.Combine('')`

Reply:
done